MsCorePkg/Library/PlatformBootManagerLib/BdsPlatform.c

```diff
@@ -41,11 +41,15 @@ ExitPmAuth (
 
   DEBUG ((DEBUG_INFO, "ExitPmAuth ()- Start\n"));
 
+  BeforeEndOfDxeEventHook (); // MS_CHANGE
+
   //
   // Since PI1.2.1, we need signal EndOfDxe as ExitPmAuth
   //
   EfiEventGroupSignal (&gEfiEndOfDxeEventGroupGuid);
 
+  AfterEndOfDxeEventHook (); // MS_CHANGE

```

Comment:
What are the usage of these hooks?

Also, is the abstraction method better than the event signaling? i.e. line 49?

@apop5 @spbrogan thoughts?

Reply:
In practical cases, we would use these hooks to send platform-specific events to enable the functionality to be activated properly.

Why I choose to use a library instead of send `EventGroupSignal`, the reason is that after the platform receives the event callback, the callback function will send its platform-specific events. I'm concerned about potential nesting of muti-events.